types/commonmark-react-renderer/index.d.ts

```diff
@@ -0,0 +1,128 @@
+
+    interface LinkProps extends CommonProps {
+        href: string;
+        title: string | undefined;
+        target?: string;
+    }
+
+    interface ImageProps extends CommonProps {
+        src: string;
+        title: string | undefined;
+        alt: string;
+    }
+
+    interface ListProps extends CommonProps {
+        start: number;
+        type: string;
+        tight: boolean;
+    }
+
+    interface Renderers {
+        Blockquote: string | React.ComponentType<CommonProps> | null;
+        Emph: string | React.ComponentType<CommonProps> | null;
+        Linebreak: string | React.ComponentType<CommonProps> | null;
+        Image: string | React.ComponentType<ImageProps> | null;
+        Item: string | React.ComponentType<CommonProps> | null;
+        Link: string | React.ComponentType<LinkProps> | null;
+        Paragraph: string | React.ComponentType<CommonProps> | null;
+        Strong: string | React.ComponentType<CommonProps> | null;
+        ThematicBreak: string | React.ComponentType<CommonProps> | null;
+        HtmlBlock: string | React.ComponentType<HtmlInlineBlockProps> | null;
+        HtmlInline: string | React.ComponentType<HtmlInlineBlockProps> | null;
+        List: string | React.ComponentType<ListProps> | null;
+        CodeBlock: string | React.ComponentType<CodeBlockProps> | null;
+        Code: string | React.ComponentType<CodeProps> | null;
+        Heading: string | React.ComponentType<HeadingProps> | null;
+        Text: string | React.ComponentType<CommonProps> | null;
+        Softbreak: string | React.ComponentType<SoftBreakProps> | null;
+    }
+
+    interface Options {
+        sourcePos?: boolean;
+        escapeHtml?: boolean;
+        skipHtml?: boolean;
+        softBreak?: string;
+        allowedTypes?: string[];
+        disallowedTypes?: string[];
+        unwrapDisallowed?: boolean;
+        allowNode?: (node: {
+            type: string;
+            renderer: string;
+            props: unknown;
+            children: unknown[];
+        }) => unknown;
+        renderers?: Partial<Renderers>;
+        transformLinkUri?: ((uri: string) => string) | null;
+        transformImageUri?: ((uri: string) => string) | null;
+        linkTarget?: string;
+    }
+}
+
+interface ReactRenderer {
+    new (options?: ReactRenderer.Options): {
+        sourcePos: boolean;
+        softBreak: string;
+        renderers: ReactRenderer.Renderers;
+        escapeHtml: boolean;
+        skipHtml: boolean;
+        transformLinUri: ((uri: string) => string) | null;
+        transformImageUri: ((uri: string) => string) | null;
+        allowNode:
+            | ((node: {
+            type: string;
+            renderer: string;
+            props: unknown;
+            children: React.ReactNode[];
+        }) => unknown)
+            | undefined;
+        allowedTypes: string[];
+        unwrapDisallowed: boolean;
+        render: (root: Node) => React.ReactNode;
+        linkTarget: string | false;
+    };
```

Comment:
```suggestion
    new (options?: ReactRenderer.Options): Required<ReactRenderer.Options>;
```

Reply:
Hey, thanks for the review! The values returned are not the same as options so this wouldn't work.

Reply:
If thats the case i recommend creating name for this interface just like options so its easier to read,

Reply:
@sheetalkamat Should be good now.